hikari/channels.py

```diff
@@ -1360,3 +1419,139 @@ class GuildStageChannel(GuildChannel):
 * `GuildTextChannel`
 * `GuildNewsChannel`
 """
+
+
+@attr.define(kw_only=True, weakref_slot=False)
+class ThreadMember:
+    """Represents a thread's member."""
+
+    thread_id: snowflakes.Snowflake = attr.field(eq=True, repr=True)
+    """ID of the thread this member is in."""
+
+    user_id: snowflakes.Snowflake = attr.field(eq=True, repr=True)
+    """The member's user ID.
+
+    !!! note
+        This will only ever be `builtins.None` on thread members attached to
+        guild create events, where this is the current bot's user.
+    """
+
+    joined_at: datetime.datetime = attr.field(eq=True, repr=True)
+    """When the user joined the relevant thread."""
+
+    flags: int = attr.field(eq=True, repr=True)
+    """Bitfield flag of the user's settings for the thread.
+
+    !!! note
+        As of writing, the values of this field's are undocumented.
+    """
+
+
+@attr.define(hash=True, kw_only=True, weakref_slot=False)
+class GuildThreadChannel(TextableGuildChannel):
```

Comment:
Guild thread channel's send method should be special cased to use parent Id as the channel id and pass self for thread Id by default